nixos/modules/services/x11/desktop-managers/default.nix

```diff
@@ -68,21 +77,18 @@ in
           scripts before forwarding the value to the
           <varname>displayManager</varname>.
         '';
-        apply = list: {
-          list = map (d: d // {
-            manage = "desktop";
-            start = d.start
-            + optionalString (needBGCond d) ''
-              if [ -e $HOME/.background-image ]; then
-                ${pkgs.feh}/bin/feh --bg-${cfg.wallpaper.mode} ${optionalString cfg.wallpaper.combineScreens "--no-xinerama"} $HOME/.background-image
-              else
-                # Use a solid black background as fallback
-                ${pkgs.xorg.xsetroot}/bin/xsetroot -solid black
-              fi
-            '';
-          }) list;
-          needBGPackages = [] != filter needBGCond list;
-        };
+        apply = map (d: d // {
+          manage = "desktop";
+          start = d.start
+          + optionalString (needBGCond d) ''
+            if [ -e $HOME/.background-image ]; then
+              ${pkgs.feh}/bin/feh --bg-${cfg.wallpaper.mode} ${optionalString cfg.wallpaper.combineScreens "--no-xinerama"} $HOME/.background-image
+            else
+              # Use a solid black background as fallback
+              ${pkgs.xorg.xsetroot}/bin/xsetroot -solid black
```

Comment:
I don't think this should have a fallback code path, in other desktop environments I noticed it being a solid black wallpaper when it should be the texture used from the window manager. Feels extra.

Reply:
Yeah probably just removing that is a better solution. I'm mainly wondering then why it's there in the first place.

Reply:
> Yeah probably just removing that is a better solution. I'm mainly wondering then why it's there in the first place.

```
  # If desktop manager `d' isn't capable of setting a background and
  # the xserver is enabled, `feh' or `xsetroot' are used as a fallback.
```

So I guess by reasoning above ^ makes this a proper feature.

It was introduced in https://github.com/NixOS/nixpkgs/commit/852813689a335399e30f83e0659666f3e312716e which is an invalid motivation IMHO. We shouldn't compensate for a desktop-manager that can't set a wallpaper in 2020??
And it conflicts with the majority that can.

Reply:
`git blame` gave me the answer: https://github.com/NixOS/nixpkgs/pull/25101

Reply:
I'm actually wondering why I ran into that in the first place.
It happened with desktopManagers that didn't use `services.xserver.desktopManager.session` to have the `bgSupport` attribute.

Reply:
So yeah, I think that PR was the wrong solution to that problem. And I can't reproduce the issue from that PR. Using the current version of the PR without the fallback branch and the following config:
```nix
{
  services.xserver = {
    enable = true;
    displayManager.sddm.enable = true;
    windowManager.i3.enable = true;
  };

  users.users.demo = {
    initialPassword = "";
    isNormalUser = true;
  };
}
```

I get the following after logging in:

![2020-01-23_05-31](https://user-images.githubusercontent.com/20525370/72956860-a8d85400-3da1-11ea-97b7-d16a77eee584.png)

Reply:
Hmm, maybe i3 started resetting the root window but I am pretty sure I encountered this issue in some DM/WM combination when testing https://github.com/NixOS/nixpkgs/pull/53843

Reply:
It was LightDM & XMonad: https://github.com/NixOS/nixpkgs/pull/53843#issuecomment-565480115